Initial block storage test

InfiniteSwerve opened the pull request:
## Problem
We want to ensure the block storage works properly. Making sure it can save and load blocks effectively, and can do so for blocks of sufficient size.

## Solution
This PR includes the minimal working example of a single empty block being saved and loaded. It also makes some necessary changes to the worker api to allow the tests to finish.

EduardoRFS commented on deku-p/src/core/stdlib/parallel.mli:
```diff
@@ -7,6 +7,7 @@ module Worker : sig
   type t = worker
 
   val make : domains:#Eio.Domain_manager.t -> sw:Eio.Switch.t -> worker
+  val teardown : worker -> unit
```
I understand why this was done, but it is quite bad as now you may have a copy of a worker that is not working, so some piece of code may try to schedule work and never receive the result.

But you can achieve the same by killing the ~sw in make, this will make everything on that switch dies and you can control granularity by making smaller switches.

InfiniteSwerve replied:
I see. I'll keep generating more tests for now like we discussed offline, but I'll make sure to address this in a future PR.

EduardoRFS commented on deku-p/src/core/block_storage/tests/block_storage_tests.ml:
```diff
@@ -0,0 +1,68 @@
+  let randn = Stdlib.Random.int 230 in
+  Deku_crypto.BLAKE2b.hash (Int.to_string randn) |> BLAKE2b.to_hex
+
+let uri = Uri.of_string (Format.sprintf "sqlite3:/tmp/%s.db" file_hash)
```
Can we have in memory database?

InfiniteSwerve replied:
Not sure. It looks like this is an option in [SQLite itself](https://www.sqlite.org/inmemorydb.html), but probably won't be available in the OCaml ecosystem.

EduardoRFS commented on deku-p/src/core/block_storage/tests/block_storage_tests.ml:
```diff
@@ -0,0 +1,68 @@
+  Parallel.Worker.teardown worker
+
+let eio_test_case : (Eio.Stdenv.t -> unit -> unit) -> unit -> unit =
+ fun f () -> Eio_main.run (fun env -> f env ())
```
you should never run Eio_main.run many times.

InfiniteSwerve replied:
Ok. As it's not a bug, and matters for performance, I'll address this in a future PR.

InfiniteSwerve commented:
@EduardoRFS I've made the changes you requested. Please review them, and if there are no blocking issues, merge this PR.

InfiniteSwerve commented:
@EduardoRFS I've rebased and, except for the in-memory sqlite database, implemented the changes you requested. Please take a look and if there are no blocking issues, merge the PR.
